src/iqm/qiskit_iqm/examples/resonance_example.py

```diff
@@ -17,31 +17,65 @@
 https://iqm-finland.github.io/qiskit-on-iqm/user_guide.html
 """
 
-from iqm.qiskit_iqm import IQMProvider, transpile_to_IQM
+import argparse
+from typing import Optional
+
 from qiskit import QuantumCircuit
 
-server_url = "https://cocos.resonance.meetiqm.com/<QUANTUM COMPUTER>"  # For example https://cocos.resonance.meetiqm.com/garnet
-api_token = "<INSERT YOUR TOKEN>"
+from iqm.qiskit_iqm import IQMProvider, transpile_to_IQM
+
+
+def resonance_example(server_url: str, api_token: Optional[str]) -> dict[str, int]:
+    """Run a circuit via IQM Resonance.
+
+    Args:
+        server_url: URL of the IQM Resonance QC used for execution
+        api_token: IQM Resonance API token for authentication
+
+    Returns:
+        a mapping of bitstrings representing qubit measurement results to counts for each result
+    """
+    SHOTS = 1000
+
+    # Define a quantum circuit
+    num_qb = 5
+    qc = QuantumCircuit(num_qb)
+
+    qc.h(0)
+    for qb in range(1, num_qb):
+        qc.cx(0, qb)
+    qc.barrier()
+    qc.measure_all()
 
-SHOTS = 1000
+    # Initialize a backend
+    backend = IQMProvider(server_url, token=api_token).get_backend()
 
-# Define a quantum circuit
-num_qb = 5
-qc = QuantumCircuit(num_qb)
+    # Transpile the circuit
+    qc_transpiled = transpile_to_IQM(qc, backend)
+    print(qc_transpiled.draw(output='text'))
 
-qc.h(0)
-for qb in range(1, num_qb):
-    qc.cx(0, qb)
-qc.barrier()
-qc.measure_all()
+    # Run the circuit
+    job = backend.run(qc_transpiled, shots=SHOTS)
+    return job.result().get_counts()
 
-# Initialize a backend
-backend = IQMProvider(server_url, token=api_token).get_backend()
 
-# Transpile the circuit
-qc_transpiled = transpile_to_IQM(qc, backend)
-print(qc_transpiled.draw(output="text"))
+if __name__ == '__main__':
+    argparser = argparse.ArgumentParser()
+    argparser.add_argument(
+        '--url',
+        help='URL of the IQM Resonance QC',
+        # For example https://cocos.resonance.meetiqm.com/garnet
+        default='https://cocos.resonance.meetiqm.com/<QUANTUM COMPUTER>',
+    )
+    argparser.add_argument(
+        '--token',
+        help='IQM Resonance access token',
+        # Provide the API token explicitly or set it as an environment variable
+        # following the https://iqm-finland.github.io/qiskit-on-iqm/user_guide.html#authentication
+        default='<INSERT YOUR TOKEN>',
+    )
 
-# Run the circuit
-job = backend.run(qc_transpiled, shots=SHOTS)
-print(job.result().get_counts())
+    args = argparser.parse_args()
+    circuit_transpiled, results = resonance_example(args.url, args.token)
+    print(circuit_transpiled)
+    print(results)
```

Comment:
`resonance_example()` doesn't return transpiled circuit, only results, so I think this should be
```suggestion
    results = resonance_example(args.url, args.token)
    print(results)
```

Reply:
(or alternatively make `resonance_example` return the transpiled circuit)

Reply:
Oh yeah, good catch, updated.